File: proj07/NameTester.rb
# NameTester.rb tests class Name and its operations
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Shion Fukuzawa, for CS 214 at Calvin College.
# Date: April 14, 2018
####################################################

require 'test/unit/assertions'   # needed for assert
include Test::Unit::Assertions

class Name 
    def initialize(first, middle, last)
        @first, @middle, @last = first, middle, last
    end

    attr_reader :first, :middle, :last

    def fullName
        @first + " " + @middle + " " + @last
    end

    def lfmi
        @last + ", " + @first + " " + @middle[0]
    end

    def read
        puts "Enter the first name: "
        firstName = gets.chomp.to_s
        puts "Enter the middle name: "
        middleName = gets.chomp.to_s
        puts "Enter the last name: "
        lastName = gets.chomp.to_s
        @first = firstName
        @middle = middleName
        @last = lastName
    end

    def print
        puts fullName
        fullName
    end
 end

def testName
   name = Name.new("John", "Paul", "Jones")

   assert name.first == "John", "first failed"
   assert name.middle == "Paul", "middle failed"
   assert name.last == "Jones", "last failed"
   assert name.fullName == "John Paul Jones", "fullName failed"
   assert name.lfmi == "Jones, John P", "lfmi failed"
   assert name.print == "John Paul Jones", "print failed"
   
   name.read()
   name.print

   print "All tests passed!\n"
end

testName

File: proj07/NameTester.java
/* NameTester.java is a 'driver' to test class Name.
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Student: Shion Fukuzawa, for CS 214 at Calvin College.
 * Date: April 14, 2018
 ************************************************************/

import java.util.Scanner;

public class NameTester  
{
  public static void main(String[] args)
  {
     Name aName = new Name("John", "Paul", "Jones");

     assert aName.getFirst().equals("John") ;
     assert aName.getMiddle().equals("Paul") ;
     assert aName.getLast().equals("Jones") ;

     System.out.println(aName);
     assert aName.toString().equals("John Paul Jones");

     aName.setFirst("Peter");
     assert aName.getFirst().equals("Peter") ;
     aName.setMiddle("Piper");
     assert aName.getMiddle().equals("Piper") ;
     aName.setLast("Smith");
     assert aName.getLast().equals("Smith") ;
     assert aName.lfmi().equals("Smith, Peter P.");

     aName.read();
     System.out.println(aName);


      System.out.println("All tests passed!");
  }
}

// Replace this line with the declaration of class Name...
class Name {
  public Name(String first, String middle, String last)
  {
    myFirst = first;
    myMiddle = middle;
    myLast = last;
  }

  public String getFirst()
  {
    return myFirst;
  }

  public String getMiddle()
  {
    return myMiddle;
  }

  public String getLast() 
  {
    return myLast;
  }

  public String toString() 
  {
    return myFirst + ' ' + myMiddle + ' ' + myLast;
  }

  public void setFirst(String first) 
  {
    myFirst = first;
  }

  public void setMiddle(String middle) 
  {
    myMiddle = middle;
  }

  public void setLast(String last)
  {
    myLast = last;
  }

  public String lfmi()
  {
    return myLast + ", " + myFirst + " " + myMiddle.charAt(0) + ".";
  }

  public void read()
  {
    Scanner keyboard = new Scanner(System.in); 
    myFirst = keyboard.next();
    myMiddle = keyboard.next();
    myLast = keyboard.next();
  }

  private String myFirst,
                 myMiddle,
                 myLast;
};
